source/server/http/admin.cc

```diff
@@ -227,6 +228,80 @@ void setHealthFlag(Upstream::Host::HealthFlag flag, const Upstream::Host& host,
     break;
   }
 }
+
+// Apply a field mask to a resource message. These have a single Any field to
+// hold the resource, and other fields with items such as name, time stamp,
+// etc. We need to provide a way to trim with field masks that makes use of Any
+// transparent, since native field masks don't work with Any. We simplify and
+// assume at most one Any message in a resource.
+// TODO(htuch): we could make field masks more powerful in future and generalize
+// this to allow arbitrary indexing through Any fields. This is pretty
+// complicated, we would need to build a FieldMask tree similar to how the C++
+// Protobuf library does this internally.
```

Comment:
Sorry for those of us that don't have an advanced degree in protobuf, is it possible to flesh out this comment a bit more to explain what this code is doing at a high level (maybe with a very small example)?

Reply:
Added.